### ISSUE TITLE
Panic on Deno.resolveDns("google.de", "ANY")

### ISSUE THREAD

**Issue by foobar27** [denoland/deno#16405]
> await Deno.resolveDns("google.de", "ANY")

============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: linux x86_64
Version: 1.26.2
Args: ["deno"]

thread 'main' panicked at 'not yet implemented', /build/deno/src/deno/ext/net/ops.rs:833:12
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**Comment by k-nasa** [comment]
It's detail.

```sh
:( % RUST_BACKTRACE=full deno
Deno 1.26.2
exit using ctrl+d, ctrl+c, or close()
> await Deno.resolveDns("google.de", "ANY")

============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos aarch64
Version: 1.26.2
Args: ["deno"]

thread 'main' panicked at 'not yet implemented', /private/tmp/deno-20221018-71507-1c6m0b9/deno/ext/net/ops.rs:833:12
stack backtrace:
   0:        0x1033a9d84 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h26781f9bfbe134a6
   1:        0x102b256b4 - core::fmt::write::h5aa0c05a9caf85df
   2:        0x1033914e4 - std::io::Write::write_fmt::h018da2c8c873b16b
   3:        0x10339d85c - std::panicking::default_hook::{{closure}}::h0239cbf25cfa6d83
   4:        0x10339d548 - std::panicking::default_hook::ha0ef000a358742e2
   5:        0x102acbae8 - deno::setup_panic_hook::{{closure}}::h26c59f69aa69ff19
   6:        0x10339e14c - std::panicking::rust_panic_with_hook::had5e29c8530a78b2
   7:        0x1033aa1a4 - std::panicking::begin_panic_handler::{{closure}}::h7852e5e8d2675e0d
   8:        0x1033aa13c - std::sys_common::backtrace::__rust_end_short_backtrace::hf05a8afc922cafbe
   9:        0x10339dc94 - _rust_begin_unwind
  10:        0x102b20f80 - core::panicking::panic_fmt::h94866529fc2e06b8
  11:        0x102b20fb8 - core::panicking::panic::h3be4af5aed488748
  12:        0x102fbb49c - <core::iter::adapters::filter_map::FilterMap<I,F> as core::iter::traits::iterator::Iterator>::next::h8f211f439730221b
  13:        0x102fb40dc - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h049becefdff1fd4d
  14:        0x102c169e8 - deno_core::runtime::JsRuntime::poll_event_loop::h5218d9dbe7aeb8ab
  15:        0x10285f64c - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h1d7cde2541704401
  16:        0x102a63018 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::had8d158db86cb358
  17:        0x102a63278 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h39333d77dd5f840c
  18:        0x102a7c8bc - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h7e17ed8b405ba767
  19:        0x102a640d8 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h3dcb1120fe5ab0cf
  20:        0x102a615e4 - deno::tools::repl::run::{{closure}}::h96c0d80f32992a17
  21:        0x102a5dd08 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hf38c35a02c5f9844
  22:        0x102858d18 - deno::main::{{closure}}::h40cc80c0b5fdd615
  23:        0x1028505dc - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hc0bb13a9cd24f491
  24:        0x10284f738 - deno_runtime::tokio_util::run_local::h3976a2e91ac2a121
  25:        0x10284ee58 - deno::main::h9519a26a4bc0e450
  26:        0x102842378 - std::sys_common::backtrace::__rust_begin_short_backtrace::h87d6446f1fc2f928
  27:        0x10284ea08 - _main
```

It seems to panic when an unsupported record type is specified.

https://github.com/denoland/deno/blob/541e241eaace6c579b6c9420dd3cfc4fd872b928/ext/net/ops.rs#L832-L833


But now, "ANY" is valid as an argument.
Since the difference with `trust_dns_proto::rr::record_type::RecordType` is born, I think that it is necessary to eliminate this.

https://github.com/bluejekyll/trust-dns/blob/main/crates/proto/src/rr/record_type.rs#L34..L123

**Comment by k-nasa** [comment]
The solution I come up with

1. return error when specify unsupported record_type
2. Add implement for record_type supported by `trust_dns_proto` 
3. Use only types supported by Deno as argument types.

I think 1 and 2 are good.

**Comment by k-nasa** [comment]
I can work on this.
But I'm not familiar with deno or DNS. I would like to discuss a solution to prevent rework after implementation.

Which do you think is better? @lucacasonato

**Comment by crowlKats** [comment]
Fixed by #17336
